test/main.js

```diff
@@ -164,6 +164,60 @@ test('custom headers', async t => {
 	await server.close();
 });
 
+test('remove custom header by extending instance', async t => {
+	const server = await createTestServer();
+	server.get('/', (request, response) => {
+		response.send(request.headers);
+		response.end();
+	});
+
+	const original = ky.create({
+		headers: {
+			rainbow: 'rainbow',
+			unicorn: 'unicorn'
+		}
+	});
+	const extended = original.extend({
+		headers: {
+			rainbow: undefined
+		}
+	});
+
+	const data = await extended(server.url).json();
+
+	t.true('unicorn' in data);
+	t.false('rainbow' in data);
+
+	await server.close();
+});
+
+test('remove custom header by extending instance (Headers instance)', async t => {
+	const server = await createTestServer();
+	server.get('/', (request, response) => {
+		response.send(request.headers);
+		response.end();
+	});
+
+	const original = ky.create({
+		headers: new Headers({
+			rainbow: 'rainbow',
+			unicorn: 'unicorn'
+		})
+	});
+	const extended = original.extend({
+		headers: {
+			rainbow: undefined
+		}
+	});
+
+	const data = await extended(server.url).json();
+
+	t.false('unicorn' in data);
```

Comment:
The test should be asserting expected behavior. As-is, it is a bit confusing.

I would prefer to use `t.true()` here and then mark the test as failing with `test.failling(...)`